src/main.py

```
@@ -0,0 +1,312 @@
+            }
+        }
+    
+    def mark_signal_acted_upon(self, signal_id: str):
+        """Mark a signal as acted upon"""
+        if signal_id in self.market_signals:
+            self.market_signals[signal_id].acted_upon = True
+            logger.info(f"Signal {signal_id} marked as acted upon")
+    
+    def export_data(self, format: str = 'json') -> str:
+        """Export system data for analysis or backup"""
+        
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)